x-pack/plugins/ml/public/application/model_management/deployment_setup.tsx

```diff
@@ -168,188 +217,390 @@ export const DeploymentSetup: FC<DeploymentSetupProps> = ({
+                      <FormattedMessage
+                        id="xpack.ml.trainedModels.modelsList.startDeployment.maxAllocationsLabel"
+                        defaultMessage="Max"
+                      />
+                    }
+                    isInvalid={!!errors.max_number_of_allocations}
+                  >
+                    <EuiFieldNumber
+                      isInvalid={!!errors.max_number_of_allocations}
+                      min={(config.adaptive_allocations?.min_number_of_allocations ?? 1) + 1}
+                      value={config.adaptive_allocations?.max_number_of_allocations}
+                      onChange={(event) => {
+                        onConfigChange({
+                          ...config,
+                          adaptive_allocations: {
+                            enabled: config.adaptive_allocations?.enabled ?? false,
+                            ...config.adaptive_allocations,
+                            max_number_of_allocations: Number(event.target.value),
+                          },
+                        });
+                      }}
+                    />
+                  </EuiFormRow>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiDescribedFormGroup>
+          ) : null}
+
+          {config.priority !== undefined ? (
+            <EuiDescribedFormGroup
+              titleSize={'xxs'}
+              title={
+                <h3>
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.modelsList.startDeployment.priorityLabel"
+                    defaultMessage="Priority"
+                  />
+                </h3>
+              }
+              description={
+                <FormattedMessage
+                  id="xpack.ml.trainedModels.modelsList.startDeployment.priorityHelp"
+                  defaultMessage="Select low priority for demonstrations where each model will be very lightly used."
+                />
+              }
+            >
+              <EuiFormRow
+                label={
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.modelsList.startDeployment.priorityLabel"
+                    defaultMessage="Priority"
+                  />
+                }
+                hasChildLabel={false}
+              >
+                <EuiButtonGroup
+                  legend={i18n.translate(
+                    'xpack.ml.trainedModels.modelsList.startDeployment.priorityLegend',
+                    {
+                      defaultMessage: 'Priority selector',
+                    }
+                  )}
+                  isFullWidth
+                  idSelected={config.priority}
+                  onChange={(optionId: string) => {
+                    onConfigChange({
+                      ...config,
+                      priority: optionId as ThreadingParams['priority'],
+                    });
+                  }}
+                  options={[
```

Comment:
Would it be better for these options to be defined outside of the component? Doesn't seem like they depend on anything in the component.